sdk/dotnet/src/Microsoft.AspNetCore.Datasync.InMemory/InMemoryRepository.cs

```diff
@@ -97,6 +97,14 @@ public IQueryable<TEntity> AsQueryable()
             return _store.Values.AsQueryable();
         }
 
+        /// <summary>
+        /// Returns an unexecuted <see cref="IQueryable{T}"/> that represents the data store as a whole.
+        /// This is adjusted by the <see cref="TableController{TEntity}"/> to account for filtering and
+        /// paging requests.
+        /// </summary>
+        /// <returns>An <see cref="IQueryable{T}"/> for the entities in the data store.</returns>
+        public Task<IQueryable<TEntity>> AsQueryableAsync() => Task.FromResult(AsQueryable());
```

Comment:
Suggest we introduce a `BaseRepository` class that inherits from IRepository and implements this as a virtual method.  The other methods in the IRepository can then be abstract (or virtual, as appropriate).

Reply:
I was thinking of using the current EfCore/InMemory Repo as BaseRepository instead of add another class.
If user need complex repository implement IRepository, else if only minimal changes are needed they can override single methods.
1st case I can fix quickly, 2nd option takes longer and I don't know when I will be able to do it.
I suggest to add async in 5.0.x and evalutate BaseRepo for 5.0.y / 6.0.0

Reply:
I think this is a breaking change, so may need to wait until the 6.0.0 release anyhow.